pkg/ingester/ingester.go

```diff
@@ -3764,6 +3763,36 @@ func (i *Ingester) ShutdownHandler(w http.ResponseWriter, _ *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
+// startReadRequest tries to start a read request and returns a readRequestState object
+// representing the state of a possibly started read request. If it wasn't possible to
+// start the read request, the causing error is returned.
+func (i *Ingester) startReadRequest() (*readRequestState, error) {
```

Comment:
Related to the comment I left above: this comment should state whether the caller should call the `finishReadRequest` after finishing.

(And I still would have preferred this to return a callback)

Reply:
I have added a callback function.